api/controllers/sitemap.js

```diff
@@ -0,0 +1,74 @@
+const defaultLog = require('winston').loggers.get('defaultLog');
+const mongoose = require('mongoose');
+
+/**
+ * Generate a dynamic sitemap for all non-static pages
+ */
+exports.getDynamicSitemap = async function (req, res) {
+    defaultLog.info('DYNAMIC SITEMAP GET');
+
+    try {
+        const Project = mongoose.model('Project');
+        const CommentPeriod = mongoose.model('CommentPeriod');
+
+        // Query for all projects that are publicly accessible
+        const projects = await Project.find({
+            read: { $in: ['public'] }
+        })
+            .select('_id name dateUpdated dateAdded')
+            .sort({ dateUpdated: -1 })
+            .lean();
+
+        // Query for published comment periods
+        const commentPeriods = await CommentPeriod.find({
+            isPublished: true
+        })
+            .select('_id project phaseName dateUpdated dateAdded')
+            .populate('project', '_id')
+            .sort({ dateUpdated: -1 })
+            .lean();
+
+        // Start building the XML sitemap
+        let sitemap = '<?xml version="1.0" encoding="UTF-8"?>\n';
+        sitemap += '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">\n';
+
+        // Add each project to the sitemap
+        projects.forEach(project => {
+            const lastModDate = project.dateUpdated || project.dateAdded || new Date();
+            const formattedDate = new Date(lastModDate).toISOString().split('T')[0];
+
+            sitemap += `  <url>\n`;
+            sitemap += `    <loc>https://planninginpartnership.ca/p/${project._id}/project-details</loc>\n`;
+            sitemap += `    <lastmod>${formattedDate}</lastmod>\n`;
+            sitemap += `    <changefreq>weekly</changefreq>\n`;
+            sitemap += `    <priority>0.8</priority>\n`;
+            sitemap += `  </url>\n`;
+        });
```

Comment:
Just wondering if we need to index the individual project page tabs, since they have their own URLs. The data could be missed by crawlers. That's p/{pid}/background-info, p/{pid}/commenting, p/{pid}/documents, p/{pid}/project-phase. If you don't think these should be indexed then that's fine too.